### PR TITLE
Fix the argumentAxis.categories description (T890311)

### PR DESCRIPTION
**Pull request by AlexJameson**

### CODE REVIEW

**Review by pavelgruba — APPROVED** [review]

**Review by pavelgruba — APPROVED** [review]

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 1]
```diff
@@ -7,54 +7,72 @@ type: Array<Number, String, Date>
...
-        { continent: 'Africa', area: 30370000 },
-        { continent: 'North America', area: 24490000 },
-        { continent: 'South America', area: 17840000 },
-        { continent: 'Antarctica', area: 13720000 },
-        { continent: 'Europe', area: 10180000 },
-        { continent: 'Australia', area: 9008500 }
-    ];
-
-To sort the arguments, for example, alphabetically, you need to assign an array of properly sorted arguments to the **categories** option.
+Arguments of the `string` type on discrete axes keep the order of objects in the data source. Assign an array of properly sorted arguments to the **categories** option to sort the arguments, for example, alphabetically:
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]:
```suggestion
Arguments of the `string` type on discrete axes keep the order of objects in the data source. Assign an array of properly sorted arguments to the **categories** option to sort the arguments as you need, for example, alphabetically: 
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 2]
```diff
@@ -7,54 +7,72 @@ type: Array<Number, String, Date>
...
 
 ##### Angular
 
-    <!--HTML--><dx-chart ... >
+    <!--HTML--><dx-chart ...
+        [dataSource]="dataSource">
         <dxo-argument-axis
-            [categories]="continents">
+            [categories]="continentNames"
+            [argumentField]="continent">
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]:
```suggestion
            argumentField="continent">
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 3]
```diff
@@ -7,54 +7,72 @@ type: Array<Number, String, Date>
...
+            'Europe',
+            'North America',
+            'South America'
+        ];
     });
 
 ##### Angular
 
-    <!--HTML--><dx-chart ... >
+    <!--HTML--><dx-chart ...
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]: The proper tab name is missing.

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 4]
```diff
@@ -95,15 +128,25 @@ To sort the arguments, for example, alphabetically, you need to assign an array
...
-                continents: [
+                dataSource: [
+                    { continent: 'Asia', area: 43820000 },
+                    { continent: 'Africa', area: 30370000 },
+                    { continent: 'North America', area: 24490000 },
+                    { continent: 'South America', area: 17840000 },
+                    { continent: 'Antarctica', area: 13720000 },
+                    { continent: 'Europe', area: 10180000 },
+                    { continent: 'Australia', area: 9008500 }
+                ];
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]: It's an object, it cannot contain semicolons.

```suggestion
                ],
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 5]
```diff
@@ -95,15 +128,25 @@ To sort the arguments, for example, alphabetically, you need to assign an array
...
+                
+                continentNames = [
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]:
```suggestion
                continentNames: [
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 6]
```diff
@@ -95,15 +128,25 @@ To sort the arguments, for example, alphabetically, you need to assign an array
...
                     'Africa', 
                     'Antarctica', 
                     'Asia', 
                     'Australia',
                     'Europe',
                     'North America',
                     'South America'
-                ]
+                ];
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]:
```suggestion
                ]
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 7]
```diff
@@ -144,4 +199,6 @@ To sort the arguments, for example, alphabetically, you need to assign an array
 
 ---
 
-[note]Arguments missing from the **categories** array will be added to its end automatically.
+Arguments of the `number` and `date` types on discrete axes are sorted in ascending order regardless of the order they have in the data source. Do not specify the **categories** option for these argument types.
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]: Is is forbidden to specify `categories` in this case, or are categories simply ignored?

File: api-reference/20 Data Visualization Widgets/dxPolarChart/1 Configuration/argumentAxis/categories.md [hunk 8]
```diff
@@ -4,7 +4,202 @@ type: Array<Number, String, Date>
...
 
 ---
-If you specify the chart's data using a common array of objects, these objects may be displayed in a random order. If you set the chart's data for each series individually, the order in which the series are positioned in the [series](/api-reference/20%20Data%20Visualization%20Widgets/dxPolarChart/1%20Configuration/series '/Documentation/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/series/') array can also be random. In these instances, the resulting argument order on the discrete argument axis may not be appropriate. To specify the order of categories (arguments on a discrete argument axis), assign an array of category names to the **categories** property.
\ No newline at end of file
+Arguments of the `string` type on discrete axes keep the order of objects in the data source. Assign an array of properly sorted arguments to the **categories** option to sort the arguments, for example, alphabetically:
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxPolarChart/1 Configuration/argumentAxis/categories.md]: I think it is possible to inherit the description from the Chart. Of course, you should use {WidgetName} in the Chart's code in this case.

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 9]
```diff
@@ -144,4 +200,6 @@ To sort the arguments, for example, alphabetically, you need to assign an array
 
 ---
 
-[note]Arguments missing from the **categories** array will be added to its end automatically.
+Arguments of the `number` and `date` types on discrete axes are sorted in ascending order regardless of the order they have in the data source.
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]: It's now unclear why we mention numeric and date types if `categories` have no impact on them or their use is not recommended.

**Review by pavelgruba — APPROVED** [review]

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 10]
```diff
@@ -4,57 +4,77 @@ type: Array<Number, String, Date>
...
-        { continent: 'North America', area: 24490000 },
-        { continent: 'South America', area: 17840000 },
-        { continent: 'Antarctica', area: 13720000 },
-        { continent: 'Europe', area: 10180000 },
-        { continent: 'Australia', area: 9008500 }
-    ];
-
-To sort the arguments, for example, alphabetically, you need to assign an array of properly sorted arguments to the **categories** option.
+Arguments of the `string` type on discrete axes keep the order of objects in the data source. Arguments of the `number` and `date` types are sorted in ascending order regardless of the order they have in the data source.
+Specify the **categories** array to set the required order of arguments. In the following example, arguments are sorted alphabetically:
```
Comment (RomanTsukanov) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]: Should this sentence be on a new line?

**Review by pavelgruba — APPROVED** [review]

**Review by RomanTsukanov — APPROVED** [review]

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md [hunk 11]
```diff
@@ -4,57 +4,76 @@ type: Array<Number, String, Date>
...
-        { continent: 'Africa', area: 30370000 },
-        { continent: 'North America', area: 24490000 },
-        { continent: 'South America', area: 17840000 },
-        { continent: 'Antarctica', area: 13720000 },
-        { continent: 'Europe', area: 10180000 },
-        { continent: 'Australia', area: 9008500 }
-    ];
-
-To sort the arguments, for example, alphabetically, you need to assign an array of properly sorted arguments to the **categories** option.
+Arguments of the `string` type on discrete axes keep the order of objects in the data source. Arguments of the `number` and `date` types are sorted in ascending order regardless of the order they have in the data source. Specify the **categories** array to set the required order of arguments. In the following example, arguments are sorted alphabetically:
```
Comment (albertov05) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/argumentAxis/categories.md]:
```suggestion
Arguments of the `string` type on discrete axes maintain the order of objects in the data source. Arguments of the `number` and `date` types are sorted in ascending order regardless of their order within the data source. Specify the **categories** array to set the required order of arguments. In the following example, arguments are sorted alphabetically: 
```

File: api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/valueAxis/categories.md [hunk 12]
```diff
@@ -6,7 +6,197 @@ type: Array<Number, String, Date>
 ##### shortDescription
 Specifies the order of categories on an axis of the *"discrete"* [type](/api-reference/20%20Data%20Visualization%20Widgets/dxChart/1%20Configuration/valueAxis/type.md '/Documentation/ApiReference/Data_Visualization_Widgets/dxChart/Configuration/valueAxis/#type').
 
+--- 
+
+Values of the `string` type on discrete axes keep the order of objects in the data source. Values of the `number` and `date` types are sorted in ascending order regardless of the order they have in the data source. Specify the **categories** array to set the required order of values. In the following example, the values are sorted alphabetically:
```
Comment (albertov05) [inline api-reference/20 Data Visualization Widgets/dxChart/1 Configuration/valueAxis/categories.md]:
```suggestion
Values of the `string` type on discrete axes maintain the order of objects in the data source. Values of the `number` and `date` types are sorted in ascending order regardless of their order within the data source. Specify the **categories** array to set the required order of values. In the following example, the values are sorted alphabetically: 
```

**Pull request merged**